push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/TokenLoader.java

```diff
@@ -134,32 +134,37 @@ public void loadAndQueueTokenBatch(@Observes @Dequeue MessageHolderWithVariants
                 String lastTokenInBatch = null;
                 int tokensLoaded = 0;
                 for (int batchNumber = 0; batchNumber < configuration.batchesToLoad(); batchNumber++) {
+
+                    // increasing the serial ID,
+                    // to make sure it's properly read from all block
+                    ++serialId;
```

Comment:
Would be nice to get @lfryc take a quick look here too, but I think the current way of increasing is wrong